### ISSUE TITLE
DL4J batch norm does not allow for 3d input

### ISSUE THREAD

**Issue by ron212** [deeplearning4j/deeplearning4j#4502]
Hii, 
I create model with Keras 2 with this code
```python
def dummy_model():
    from keras.models import Sequential
    from keras.layers import Dense, Dropout, Conv1D, Flatten
    from keras.layers.normalization import BatchNormalization
    from keras.layers.pooling import MaxPooling1D
    p = 0.3
    model = Sequential([
        BatchNormalization(input_shape=(10, 20)),  # axis=-1
        Conv1D(16, kernel_size=5, padding='same', activation='relu'),
        MaxPooling1D(pool_size=2),
        Flatten(),
        Dense(32, activation='relu'),
        Dropout(p),
        Dense(1, activation='sigmoid')
    ])
    model.compile(loss='binary_crossentropy', optimizer='adam', metrics=['accuracy'])  # 'rmsprop'
    model.summary(print_fn=logging.debug)
    x = np.random.random((100,10,20))
    y = np.random.choice(1, 100)
    model.fit(x, y, batch_size=16, epochs=10)
    model.save_weights("/tmp/nagar/model_save_weights")
    json_str = model.to_json()
    with open("/tmp/nagar/model_json", "w") as f:
        f.write(json_str)
```
i would like to run prediction with DL4J, this is the snippet i use
```java
MultiLayerNetwork model =
      KerasModelImport.importKerasSequentialModelAndWeights("model_json",  "model_save_weights")
```
and i get this error
```
[error] (run-main-0) java.lang.IllegalStateException: Invalid input type: Batch norm layer expected input of type CNN, CNN Flat or FF, got InputTypeRecurrent(20) for layer index -1, layer name = batch_normalization_1
[error] java.lang.IllegalStateException: Invalid input type: Batch norm layer expected input of type CNN, CNN Flat or FF, got InputTypeRecurrent(20) for layer index -1, layer name = batch_normalization_1
[error]     at org.deeplearning4j.nn.conf.layers.BatchNormalization.getOutputType(BatchNormalization.java:96)
[error]     at org.deeplearning4j.nn.modelimport.keras.layers.normalization.KerasBatchNormalization.getOutputType(KerasBatchNormalization.java:124)
[error]     at org.deeplearning4j.nn.modelimport.keras.KerasModel.inferOutputTypes(KerasModel.java:262)
[error]     at org.deeplearning4j.nn.modelimport.keras.KerasSequentialModel.<init>(KerasSequentialModel.java:129)
[error]     at org.deeplearning4j.nn.modelimport.keras.KerasSequentialModel.<init>(KerasSequentialModel.java:60)
[error]     at org.deeplearning4j.nn.modelimport.keras.utils.KerasModelBuilder.buildSequential(KerasModelBuilder.java:120)
[error]     at org.deeplearning4j.nn.modelimport.keras.KerasModelImport.importKerasSequentialModelAndWeights(KerasModelImport.java:237)
```

**Comment by maxpumperla** [comment]
@ron212 thanks for reporting, this is pretty much the same as https://github.com/deeplearning4j/deeplearning4j/issues/4466, but I leave both open for now. will look into it.

**Comment by tkocmathla** [comment]
@maxpumperla I also get this error when importing a Keras model that goes Conv1D -> MaxPooling1D -> BatchNormalization. The error message is identical to @ron212's except for the dimensionality.

I noticed you fixed the related issue #4466 -- is there a similar fix that can be applied here? If there's anything more I can provide to help, let me know.

**Comment by tkocmathla** [comment]
I was able to reproduce this with a minimal test case:
```
in_layer = Input((25, 25,))
x = Conv1D(filters=10, kernel_size=2)(in_layer)
out_layer = BatchNormalization()(x)
model = Model(in_layer, out_layer)

with open('simple_model.json', 'w') as f:
    f.write(model.to_json())
```

Then, in Clojure:
```
(KerasModelImport/importKerasModelConfiguration "simple_model.json")
```

The exception thrown says:
```
java.lang.IllegalStateException: Invalid input type: Batch norm layer expected input of type CNN, CNN Flat or FF, got InputTypeRecurrent(10) for layer index -1, layer name = batch_normalization_7
```

**Comment by maxpumperla** [comment]
@ron212 @tkocmathla  thanks, the root cause for this is not an import problem but rather that DL4J  batch norm does simply not support this right now. I'll try to make this happen, but it's a little intricate. not sure I can squeeze this in before the new release. we'll see.

**Comment by tkocmathla** [comment]
What's odd (to me) is that I can build a similar architecture directly in DL4J (Conv1D -> BatchNorm). I'm brand new to DL4J, so this may not be a sufficient example, but on the surface it seems like the connections are the same here, and it "works":

```clojure
(->> (doto
       (ComputationGraph.
         (.. (NeuralNetConfiguration$Builder.)
             (graphBuilder)
             (addInputs (into-array ["in"]))
             (addLayer "conv" (.. (Convolution1DLayer$Builder. 3) (nIn 100) (nOut 100) build) (into-array ["in"]))
             (addLayer "norm" (.. (BatchNormalization$Builder.) (nIn 100) (nOut 100) build) (into-array ["conv"]))
             (addLayer "out" (.. (OutputLayer$Builder.) (nIn 100) (nOut 10) build) (into-array ["norm"]))
             (setOutputs (into-array ["out"]))
             build))
       .init)
     .summary
     println)
```

**Comment by raver119** [comment]
Our new batchnorm available in libnd4j should support 3D inputs i think.
